plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -984,7 +985,12 @@ func (c *PodAffinityChecker) InterPodAffinityMatches(pod *v1.Pod, meta interface
 	if node == nil {
 		return false, nil, fmt.Errorf("node not found")
 	}
-	if !c.satisfiesExistingPodsAntiAffinity(pod, meta, node) {
+	allPods, err := c.podLister.List(labels.Everything())
```

Comment:
Can we add a metadata precomputation here as this seems to be an expensive operation just like we have metadata for checkServiceAffinity?